projects/core/utils/dom/process-icon.ts

```diff
@@ -7,8 +7,8 @@ export function processIcon(source: string, name: string): string {
         return source;
     }
 
-    const src = source.substring(source.indexOf(START));
-    const attributes = src.substring(0, src.indexOf('>'));
+    const src = source.slice(Math.max(0, source.indexOf(START)));
+    const attributes = src.slice(0, Math.max(0, src.indexOf('>')));
```

Comment:
`substring` doesn't seem deprecated and it doesn't require extra `Math.max`, are u sure about replacing?

Reply:
`String#substr() and String#substring() are the two lesser known legacy ways to slice a string. It's better to use String#slice() as it's a more popular option with clearer behavior that has a consistent Array counterpart.`

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-slice.md